### PR TITLE
ADR seperating meteor builder and pipelines

### PR DESCRIPTION
**Pull request by goern**
Here is a proposal how to solve https://github.com/AICoE/meteor/issues/90 and https://github.com/AICoE/aicoe-ci/issues/127

Signed-off-by: Christoph Görn <goern@redhat.com>

### CODE REVIEW

File: docs/adr/000-refactor-pipelines.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Refactor all Build and Deployment Pipelines into external component
+
+* Status: accepted
+* Deciders:
+* Date: 2021-09-15
+
+## Context and Problem Statement
+
+Meteor builder is ment as an orchestrator build and deploy tasks. Embedding the build and deploy logic itself into meteor or
+meteor-operator introduces a tight compling of components that makes independend development and code reuse harder.
```
Comment (codificat) [inline docs/adr/000-refactor-pipelines.md:10]:
```suggestion
Meteor builder is ment as an orchestrator build and deploy tasks. Embedding the build and deploy logic itself into meteor or
meteor-operator introduces a tight compling of components that makes independend development and code reuse harder.
```
```suggestion
Meteor builder is meant as an orchestrator for build and deploy tasks. Embedding the build and deploy logic itself into meteor or
meteor-operator introduces a tight coupling of components that makes independent development and code reuse harder.
```

File: docs/adr/000-refactor-pipelines.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Refactor all Build and Deployment Pipelines into external component
+
+* Status: accepted
+* Deciders:
```
Comment (codificat) [inline docs/adr/000-refactor-pipelines.md:4]: should be populated

**Review by codificat — CHANGES_REQUESTED** [review]
We should get an agreement on this. @harshad16 @tumido thoughts?

My two cents: looks good to me, only a couple of typos to fix.

Also adding a note here that, if accepted, it:
Fixes AICoE/aicoe-ci#127
(by making it obsolete)

**Comment by goern** [comment]
@tumido bounce

File: docs/adr/000-refactor-pipelines.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## Decision Drivers
+
+Build and deployment of container images and applications is a utility that could be reused in different scenarios, therefore
+a separation of these is advised.
+
+## Considered Options
+
+1. Tekton YAML as port of meteor-operator
+2. Tekton YAML as a seperate Helm application
+3. Tekton YAML as a seperate GitHub with Kustomize manifests
```
Comment (harshad16) [inline docs/adr/000-refactor-pipelines.md:24]:
```suggestion
3. Tekton YAML as a seperate GitHub repo with Kustomize manifests
```

**Review by harshad16 — COMMENTED** [review]
lgtm
we can move the core pipelines into the helm charts along with the tasks  :+1:

File: docs/adr/000-refactor-pipelines.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+### Positive Consequences
+
+* Tekton YAML can be developed and distributed independently of Meteor builder.
+* Python Module Build/Delivery Pipelines can be deployed independently of Meteor builder.
+* all build/deliver/deploy related components are maintained in one place
+
+### Negative Consequences
+
+* a stable interface between two projects need to be maintained
+* ArgoCD needs to deploy Helm
```
Comment (tumido) [inline docs/adr/000-refactor-pipelines.md:41]: This is not true. There's a tech preview Helm deployment in developer perspective in OCP and ACM also offers a `HelmRelease` resource kind, we can make this fully declarative. This negative consequence is not valid. :raised_hands:

**Review by tumido — COMMENTED** [review]
besides already pointed out changes in previous reviews, I agree with this.

File: docs/adr/000-refactor-pipelines.md (hunk 4 above)
Comment (goern) [reply docs/adr/000-refactor-pipelines.md]: please check https://github.com/AICoE/meteor/pull/154/commits/473884f3cc156550ea6ce3421a77ef3d19534598

**Review by codificat — CHANGES_REQUESTED** [review]
Just a typo fix.

/lgtm

**Comment by goern** [comment]
ping @tumido @codificat

**Comment by codificat** [comment]
/lgtm
thanks!

/assign @tumido 
for final /approve

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**